caikit_nlp/modules/token_classification/transformer_sentence_classification.py

```diff
@@ -0,0 +1,218 @@
+from .token_classification_task import TokenClassificationTask
+
+log = alog.use_channel("SEQ_TSFMR")
+error = error_handler.get(log)
+
+
+@module(
+    id="42a7d920-8b7e-4e1f-81fb-8ab851a80c99",
+    name="Transformer sentence-level classification",
+    version="0.1.0",
+    task=TokenClassificationTask,
+)
+class TransformerSentenceClassification(ModuleBase):
+
+    ################################ Constructor #################################################
+
+    def __init__(
+        self,
+        lang: str,
+        sentence_splitter: SentenceSplitBase,
+        sequence_classifier: SequenceClassification,
+        default_threshold: float,
+        labels_to_output: List[str] = None,
+        labels_mapping: Dict[str, str] = None,
+    ):
+        """Construct a transformer sentence classification object
+        from a sentence splitter and sequence classifier
+
+        Args:
+            lang: str
+                2 letter language code
+            sentence_splitter: SentenceSplitBase
+                Sentence splitter that returns List[Span]
+            sequence_classifier: SequenceClassification
+                Sequence tokenizer and classification model
+            default_threshold: float
+                Default threshold for scores
+            labels_to_output: List[str]
+                (Optional) Labels to output
+            labels_mapping: Dict[str, str]
+                (Optional) Mapping of model labels to more semantically meaningful labels
+        """
+        super().__init__()
+        error.type_check("<NLP12578168E>", str, lang=lang)
+        error.type_check(
+            "<NLP79642537E>", SentenceSplitBase, sentence_splitter=sentence_splitter
+        )
+        error.type_check(
+            "<NLP35742128E>",
+            SequenceClassification,
+            sequence_classifier=sequence_classifier,
+        )
+        error.type_check("<NLP63802045E>", float, default_threshold=default_threshold)
+        error.type_check_all(
+            "<NLP71653678E>", str, allow_none=True, labels_to_output=labels_to_output
+        )
+        error.type_check(
+            "<NLP56932573E>", Dict, allow_none=True, labels_mapping=labels_mapping
+        )
+        self.lang = lang
+        self.sentence_splitter = sentence_splitter
+        self.sequence_classifier = sequence_classifier
+        self.default_threshold = default_threshold
+        self.labels_to_output = labels_to_output
+        self.labels_mapping = labels_mapping
+
+    ################################## API functions #############################################
+
+    def run(
+        self, text: str, threshold: Optional[float] = None
+    ) -> TokenClassificationResult:
+        """Run sentence-level classification on text. Returns results
+        based on score threshold for labels that are to be outputted
+
+        Args:
+            text: str
+                Document to run sentence-level classification on
+            threshold: float
```

Comment:
I think the thresholding concepts raises some questions:
1. How will a user know what is the right threshold or even the range (upper and lower limit of the score)
2.. Would they not receive any results that are of score lower than the threshold and would the response look empty in that case
3. Would this threshold me universal or change based on model.

Reply:
> Would this threshold me universal or change based on model.

The `default_threshold` would be stored with the model so changeable based on model (model author can define their concept). Thus "How will a user know what is the right threshold or even the range" is going to be model dependent and require model-specific documentation.

The override here at runtime is merely for more flexibility 

>Would they not receive any results that are of score lower than the threshold and would the response look empty in that case

I think the response looking empty in the non-streaming case is fine i.e. no X is found for whatever labels or all labels that are being outputted. For streaming (not in this PR) we will consider returning "everything"/all the input to indicate that there was processing done vs. hanging response